apps/dashboard/src/components/workflow-editor/workflow-provider.tsx

```diff
@@ -0,0 +1,177 @@
+import { PatchWorkflowDto, UpdateWorkflowDto, WorkflowResponseDto } from '@novu/shared';
+import { createContext, ReactNode, useLayoutEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { ToastIcon } from '@/components/primitives/sonner';
+import { showToast } from '@/components/primitives/sonner-helpers';
+import { useEnvironment } from '@/context/environment/hooks';
+import { useFetchWorkflow, useUpdateWorkflow } from '@/hooks';
+import { useDebounce } from '@/hooks/use-debounce';
+import { usePatchWorkflow } from '@/hooks/use-patch-workflow';
+import { createContextHook } from '@/utils/context';
+import { buildRoute, ROUTES } from '@/utils/routes';
+
+export type WorkflowContextType = {
+  isPending: boolean;
+  workflow?: WorkflowResponseDto;
```

Comment:
Consider making the `workflow` DTO or null (when not loaded) instead of optional.